psamm/command.py

```diff
@@ -91,6 +92,16 @@ def init_parser(cls, parser):
     def run(self):
         """Execute command"""
 
+    def argument_error(self, msg):
+        """Raise error indicating error parsing an argument."""
+        raise CommandError(msg)
+
+    def fail(self, exc=None):
```

Comment:
No `exc` should always be an `Exception`. Otherwise the log message doesn't make any sense. You can add another parameter with a string error message that you log to `logger.error` before logging the `exc`.